packages/frontend/src/pages/Recipe.svelte

```diff
@@ -15,21 +15,31 @@ import ContainerConnectionStatusInfo from '../lib/notification/ContainerConnecti
 import { modelsInfo } from '../stores/modelsInfo';
 import { checkContainerConnectionStatus } from '../utils/connectionUtils';
 import { router } from 'tinro';
+import { InferenceType } from '@shared/src/models/IInference';
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
 
 export let recipeId: string;
 
 // The recipe model provided
 $: recipe = $catalog.recipes.find(r => r.id === recipeId);
 $: categories = $catalog.categories;
+
+// model selected to start the recipe
 let selectedModelId: string;
-$: selectedModelId = recipe?.models?.[0] ?? '';
+$: selectedModelId = recipe?.recommended && recipe.recommended.length > 0 ? recipe?.recommended?.[0] : '';
+
 let connectionInfo: ContainerConnectionInfo | undefined;
 $: if ($modelsInfo && selectedModelId) {
   checkContainerConnectionStatus($modelsInfo, selectedModelId, 'recipe')
     .then(value => (connectionInfo = value))
     .catch((e: unknown) => console.log(String(e)));
 }
 
+let models: ModelInfo[];
+$: models = $catalog.models.filter(
+  model => (model.backend ?? InferenceType.NONE) === (recipe?.backend ?? InferenceType.NONE),
```

Comment:
Wondering if we should not also consider recommanded as a way to associate a model with a recipe (old style) eg recipe without backend

Reply:
Oh yeah, that's very smart ! For the object detection that would works

Reply:
I would delay this in another PR, probably https://github.com/containers/podman-desktop-extension-ai-lab/issues/1126 as this would probably need some UI/UX because we would still want user to be able to do their own choice I guess